### PR TITLE
[WIP] Adds code review notes about migration concerns

### PR DESCRIPTION
**Pull request by cbortz**
> Pivotal Tracker
> https://www.pivotaltracker.com/story/show/152620739

WIP

### CODE REVIEW

**Comment by cbortz** [comment]
@dleve123 I apologize for this taking so long to get rolling. Please see the suggested edits and let's get this conversation going.

File: code-review/README.md [hunk 1]
```diff
@@ -84,6 +84,36 @@ experience, refactors the existing code). Then:
...
+
+Due to our deployment process, we can encounter momentary `500` statuses when performing certain migration procedures. Namely, **removing** and **renaming** columns can result in these intermittent downtimes. Here's an example:
+
+1. We have an attribute `Model#unused_attribute`.
+2. A request comes in to remove this attribute and all references.
+3. We generate a migration to remove the attribute and references.
+4. We deploy the changes.
+5. _Uh oh..._
+
+During the migration period, just after the column is removed, the codebase changes have yet to rollover. Thus, if a user visits a page that references `Model#unused_attribute`, an exception will be raised, resulting in some potential downtimes.
```
Comment (dleve123) [inline code-review/README.md:97]: Hmm, I believe the order of operations is reversed, right?

1. Deploy
2. Migrate (via the `before_release` hook defined in .aptible.yml)

The issue happens between (1) and (2), if I'm understanding correctly

File: code-review/README.md [hunk 2]
```diff
@@ -84,6 +84,36 @@ experience, refactors the existing code). Then:
 * Sign off on the pull request by approving a [pull request
   review][pr-review-docs] in Github.
 
+### Migrations & Potential Downtime
+
+Due to our deployment process, we can encounter momentary `500` statuses when performing certain migration procedures. Namely, **removing** and **renaming** columns can result in these intermittent downtimes. Here's an example:
```
Comment (dleve123) [inline code-review/README.md:89]: IMO, `intermittent downtimes` reads a bit weird, `intermittent downtime` (singular) reads better

**Review by dleve123 — COMMENTED** [review]
@cbortz with the exception of a few comments, I agree with the content of this documentation.

However, I wonder what the plan is to operationalize this information - is there a way we can configure Hound CI to comment on non-pure-additive diffs to tables?

In other words, short of expecting developers to keep this information in their heads when developing and code reviewing, what can we do to systematize this guidance?

File: code-review/README.md (hunk 1 above)
Comment (cbortz) [reply code-review/README.md:97]: @dleve123 "Deploy" (step 4) in this context means push to the Aptible remote, which migrates before rolling over the new code. I can update this to provide better clarity.

**Comment by KrishnaKulkarni** [comment]
@cbortz What's the status of this?

**Comment by cbortz** [comment]
@KrishnaKulkarni Waiting on a response to my clarification on https://github.com/healthify/guides/pull/23#discussion_r155104232

After that a quick fixup should have this ready to be merged.

File: code-review/README.md (hunk 1 above)
Comment (dleve123) [reply code-review/README.md:97]: Updating sounds good!

**Pull request closed without merging**